packages/client/test/integration/ProxyPublishing.test.ts

```diff
@@ -0,0 +1,160 @@
+import { createTestStream, fakePrivateKey, getCreateClient } from '../utils'
+import { ConfigTest, Stream, StreamOperation, StreamrClient } from '../../src'
+import { wait } from 'streamr-test-utils'
+import { SPID } from 'streamr-client-protocol'
+
+describe('PubSub with proxy connections', () => {
+    let stream: Stream
+    let publishingClient: StreamrClient
+    let proxyClient1: StreamrClient
+    let proxyClient2: StreamrClient
+    let pubPrivateKey: string
+    let proxyPrivateKey1: string
+    let proxyPrivateKey2: string
+    let proxyNodeId1: string
+    let proxyNodeId2: string
+    const createClient = getCreateClient()
+
+    beforeEach(async () => {
+        pubPrivateKey = fakePrivateKey()
+        proxyPrivateKey1 = fakePrivateKey()
+        proxyPrivateKey2 = fakePrivateKey()
+
+        publishingClient = createClient({
+            id: 'publisher',
+            auth: {
+                privateKey: pubPrivateKey
+            }
+        })
+        proxyClient1 = createClient({
+            id: 'proxy',
+            auth: {
+                privateKey: proxyPrivateKey1
+            },
+            network: {
+                acceptProxyConnections: true,
+                trackers: ConfigTest.network.trackers
+            }
+        })
+        proxyClient2 = createClient({
+            id: 'proxy',
+            auth: {
+                privateKey: proxyPrivateKey2
+            },
+            network: {
+                acceptProxyConnections: true,
+                trackers: ConfigTest.network.trackers
+            }
+        })
+        proxyNodeId1 = await proxyClient1.node.getNodeId()
+        proxyNodeId2 = await proxyClient2.node.getNodeId()
+        stream = await createTestStream(publishingClient, module)
+        const pubUser = await publishingClient.getUserInfo()
+        const proxyUser = await proxyClient1.getUserInfo()
+        const proxyUser2 = await proxyClient2.getUserInfo()
+
+        await stream.grantPermission(StreamOperation.STREAM_GET, pubUser.username)
+        await stream.grantPermission(StreamOperation.STREAM_PUBLISH, pubUser.username)
```

Comment:
There's a new client api that allows you to set multiple permissions at once:

```suggestion
        await stream.grantPermissions([StreamOperation.STREAM_PUBLISH, StreamOperation.STREAM_PUBLISH] pubUser.username)
```